lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -35,15 +38,50 @@ import { replaceWindowsNetworkDriveLetter } from '../../os/windows-network-drive
 import {
 	ChangeButton,
 	DetailsText,
-	Footer,
 	StepButton,
 	StepNameButton,
 	StepSelection,
-	Underline,
 } from '../../styled-components';
+import { colors } from '../../theme';
 import { middleEllipsis } from '../../utils/middle-ellipsis';
 import { SVGIcon } from '../svg-icon/svg-icon';
 
+const jsonStorage = {
+	get: (key: string) => {
+		return new Promise((resolve, reject) => {
+			jsonStorageCb.get(key, (err, value) => {
+				if (err) {
+					reject(err);
+					throw err;
+				}
+				resolve(value);
+				return value;
+			});
+		});
+	},
```

Comment:
Throwing or returning in a `new Promise((resolve, reject) => { ... })` makes no sense.
Use `util.promisify`:
```typescript
const jsonStorage = {
    get: promisify(jsonStorageCb.get),
    set: promisify(jsonStorageCb.set),
}
```

Reply:
Actually just use `localStorage.setItem('recentUrlImages', JSON.stringify(yourData))` and `JSON.load(localStorage.getItem('recentUrlImages'))` and remove electron-json-storage completely